alert-database/src/main/java/com/synopsys/integration/alert/database/job/DistributionJobRepository.java

```diff
@@ -51,7 +51,7 @@
                        + "    AND blackDuckDetails.globalConfigId = :blackDuckConfigId"
                        + "    AND notificationTypes.notificationType IN (:notificationTypeSet)"
                        + "    AND jobEntity.distributionFrequency IN (:frequencies)"
-                       + "    AND (blackDuckDetails.filterByProject = false OR blackDuckDetails.projectNamePattern IS NOT NULL OR projects.projectName IN (:projectNames))"
+                       + "    AND (blackDuckDetails.filterByProject = false OR blackDuckDetails.projectNamePattern IS NOT NULL OR blackDuckDetails.projectVersionNamePattern IS NOT NULL OR projects.projectName IN (:projectNames))"
```

Comment:
I don't think we want `blackDuckDetails.projectVersionNamePattern IS NOT NULL` as part of the query. Since we are not comparing the version in the query, it does not matter if it is set or not at this point.

Reply:
If only the project version name pattern is set (no project selected and no project name pattern), this query will not properly read jobs from the DB. There is also a test to verify this in JobNotificationMapperTestIT.

Reply:
If we can search by projectVersionNamePattern only then we need to treat it like projectNamePattern and filterByProject is false.  We need to include all the projects and then filter using the project version name pattern in java code just like the projectNamePattern.  So I think that is needed.

Reply:
We're just trying to find the jobs that might match the notification so projectVersionNamePattern not null might match the notification so we want to include the job in the results.